### PR TITLE
Terraform Example Code Optimized

### PR DESCRIPTION
**Pull request by marifse**
### Release required?

- [ ] No release
- [ ] Patch release (`x.x.X`)
- [ ] Minor release (`x.X.x`)
- [ ] Major release (`X.x.x`)

### Run the pipeline

If the CI pipeline doesn't run when you create the PR, the PR requires a user with GitHub collaborators access to run the pipeline.

Run the CI pipeline when the PR is ready for review and you expect tests to pass. Add a comment to the PR with the following text:

```
/run pipeline
```

### Checklist for reviewers

- [ ] If relevant, a test for the change is included or updated with this PR.
- [ ] If relevant, documentation for the change is included or updated with this PR.

### For mergers

- Use a conventional commit message to set the release level. Follow the [guidelines](https://terraform-ibm-modules.github.io/documentation/#/merging.md).
- Include information that users need to know about the PR in the commit message. The commit message becomes part of the GitHub release notes.
- Use the **Squash and merge** option.

### CODE REVIEW

**Comment by marifse** [comment]
@ocofaigh can you please have a look at this pull request, and let me know your feedback.

**Comment by marifse** [comment]
@ocofaigh please review it too. Thanks

**Comment by marifse** [comment]
@ocofaigh can you please look into this, and approve if all okay.

File: examples/sg-target-example/catalogValidationValues.json.template [hunk 1]
```diff
@@ -0,0 +1,8 @@
+{
+  "ibmcloud_api_key": $VALIDATION_APIKEY,
+  "region": "us-south",
+  "resource_tags": $TAGS,
+  "prefix": $PREFIX,
+  "vpc_name": $VPC_NAME,
+  "zone": $ZONE
+}
```
Comment (ocofaigh) [inline examples/sg-target-example/catalogValidationValues.json.template:8]: please delete this file

File: examples/default/catalogValidationValues.json.template [hunk 2]
```diff
@@ -0,0 +1,7 @@
+{
+  "ibmcloud_api_key": $VALIDATION_APIKEY,
+  "region": "us-south",
+  "resource_tags": $TAGS,
+  "prefix": $PREFIX,
+  "vpc_name": $VPC_NAME
+}
```
Comment (ocofaigh) [inline examples/default/catalogValidationValues.json.template:7]: please delete this file

**Comment by ocofaigh** [comment]
@marifse I left 2 comments

**Comment by ocofaigh** [comment]
@marifse Test failed with:
```
TestRunDefaultExampleWithoutIBMRules 2024-03-26T15:50:47Z retry.go:99: Returning due to fatal error: FatalError{Underlying: error while running command: exit status 1; ╷
│ Error: No value for required variable
│ 
│   on variables.tf line 99:
│   99: variable "vpc_name" {
│ 
│ The root module input variable "vpc_name" is not set, and has no default
│ value. Use a -var or -var-file command line argument to provide a value for
│ this variable.
╵}
```

**Comment by ocofaigh** [comment]
@marifse Just to be clear, you wont need to add any `catalogValidationValues.json.template` files if they don't already exist. If they do exist, just ensure that all required variables for the example are in them.

**Comment by marifse** [comment]
> @marifse Just to be clear, you wont need to add any new `catalogValidationValues.json.template` files if they don't already exist. If they do exist, just ensure that all required variables for the example are in them.

@ocofaigh done

File: examples/default/main.tf [hunk 3]
```diff
@@ -22,10 +20,10 @@ module "vpc" {
   count             = var.vpc_id != null ? 0 : 1
   source            = "terraform-ibm-modules/landing-zone-vpc/ibm"
   version           = "7.13.2"
-  resource_group_id = var.resource_group != null ? data.ibm_resource_group.existing_resource_group[0].id : ibm_resource_group.resource_group[0].id
+  resource_group_id = module.resource_group.resource_group_id
   region            = var.region
   prefix            = var.prefix
-  name              = var.vpc_name
+  name              = "${var.prefix}-vpc"
```
Comment (ocofaigh) [inline examples/default/main.tf:26]: dont use the prefix here too. set it to `name = "vpc"`

File: examples/sg-target-example/main.tf [hunk 4]
```diff
@@ -23,10 +21,10 @@ module "vpc" {
   count             = var.vpc_id != null ? 0 : 1
   source            = "terraform-ibm-modules/landing-zone-vpc/ibm"
   version           = "7.13.2"
-  resource_group_id = var.resource_group != null ? data.ibm_resource_group.existing_resource_group[0].id : ibm_resource_group.resource_group[0].id
+  resource_group_id = module.resource_group.resource_group_id
   region            = var.region
   prefix            = var.prefix
-  name              = var.vpc_name
+  name              = "${var.prefix}-vpc"
```
Comment (ocofaigh) [inline examples/sg-target-example/main.tf:27]: dont use the prefix here too. set it to `name = "vpc"`

**Review by ocofaigh — CHANGES_REQUESTED** [review]
see comments

**Comment by marifse** [comment]
@ocofaigh fixed, please check now

**Comment by ocofaigh** [comment]
@marifse 
image

**Comment by marifse** [comment]
@ocofaigh please check now and rerun. conflicts resolved

**Comment by marifse** [comment]
@ocofaigh please check now and rerun. conflicts resolved

**Review by ocofaigh — APPROVED** [review]

**Pull request merged**